src/scripts/data_file_processing/healthsystem/equipment/equipment_availability_estimation.py

```diff
@@ -0,0 +1,273 @@
+levels_4_and_5_lookup = mfl.loc[mfl['Facility_Level'].isin(['4', '5'])].set_index('Facility_Level')['Facility_ID'].to_dict()
+mask_to_levels_4_and_5 = final_equipment_availability_export['Facility_Level'].isin(['4', '5'])
+final_equipment_availability_export.loc[mask_to_levels_4_and_5, 'Facility_ID'] = \
+    final_equipment_availability_export.loc[mask_to_levels_4_and_5, 'Facility_Level'].map(levels_4_and_5_lookup)
+
+assert final_equipment_availability_export['Facility_ID'].notnull().all()
+final_equipment_availability_export['Facility_ID'] = final_equipment_availability_export['Facility_ID'].astype(int)
+
+# - Format as a pd.Series with multi-index (Facility_ID, Item_Code)
+final_equipment_availability_export = final_equipment_availability_export.set_index(['Facility_ID', 'Item_Code'])['Pr_Available']
+
+# Extrapolate so that there is an estimated availability for every item at every level
+# - Create "full" dataset, where we force that there is probability of availability for every item_code at every
+#  observed facility
+_all_fac_ids = mfl.Facility_ID.unique()
+_all_item_codes = pd.read_csv(
+    resourcefilepath /'healthsystem/infrastructure_and_equipment/ResourceFile_EquipmentCatalogue.csv'
+)['Item_Code'].unique()
+final_equipment_availability_export_full = pd.Series(
+    index=pd.MultiIndex.from_product(
+        [_all_fac_ids, _all_item_codes],
+        names=["Facility_ID", "Item_Code"]
+    ),
+    data=float("nan"),
+    name='Pr_Available'
+).combine_first(final_equipment_availability_export)
+
+print(f''
+      f'Fraction of missing data when requiring data for all items at all facilities = '
+      f'{round(100 * final_equipment_availability_export_full.isnull().sum() / len(final_equipment_availability_export_full))}'
+      f'%')
+
+# Impute availability of missing item_codes in a facility_id... from the mean availability of the same items in other facilities of the same level
+facility_levels = final_equipment_availability_export_full.index.get_level_values('Facility_ID').map(
+    mfl.set_index('Facility_ID')['Facility_Level'].to_dict()
+)
+final_equipment_availability_export_full = final_equipment_availability_export_full.groupby(['Item_Code', facility_levels]).transform(lambda x: x.fillna(x.mean()))
+
+# Remaining misising data include those item_codes that are never seen anywhere: interpolate these from the average availability of all other items, stratified by level
+final_equipment_availability_export_full = final_equipment_availability_export_full.groupby([facility_levels]).transform(lambda x: x.fillna(x.mean()))
+
+# Remaining missing data are for items_codes in facility_ids for which there is no information at all in the facility_level
+# Impute the availability for these items in facilities as ... the average of other items available at other facilities
+final_equipment_availability_export_full = final_equipment_availability_export_full.groupby("Item_Code").transform(lambda x: x.fillna(x.mean()))
```

Comment:
On the final level of interpolation which is done based on 'Item_code' alone - This is interpolating values only for facility level 5, where there is no service delivery taking place. I assume this is needed to be done for the assertions in `equipment.py` to work?  
I wonder if more reasonable options might be - 
1. Either we just drop level 5 completely from  `ResourceFile_Equipment_Availability_Estimates.csv`
2. Or we force all `Pr_Available` = 0 for`Facility_Level` == 5

Reply:
Thanks Tim. I've pushed a change to implement option 2.